automation/check-patch.e2e-lifecycle-k8s.sh

```diff
@@ -37,7 +37,7 @@ main() {
     make cluster-up
     trap teardown EXIT SIGINT SIGTERM SIGSTOP
     make cluster-operator-push
-    make E2E_TEST_EXTRA_ARGS="-ginkgo.noColor --junit-output=$ARTIFACTS/junit.functest.xml" test/e2e/lifecycle
+    make E2E_TEST_EXTRA_ARGS="-ginkgo.noColor --ginkgo.junit-report=$ARTIFACTS/junit.functest.xml" test/e2e/lifecycle
```

Comment:
Move the $ARTIFACTS to --output-dir, it does not work if the dir is passed as --junit-report

Reply:
Done

Reply:
I checked using `--ginkgo.junit-report=$ARTIFACTS/junit.functest.xml"` (+failing the test on purpose) and we do see the artifacts, see [here](https://gcsweb.ci.kubevirt.io/gcs/kubevirt-prow/pr-logs/pull/kubevirt_cluster-network-addons-operator/1260/pull-e2e-cluster-network-addons-operator-workflow-k8s/1503705101607374848/)

Reply:
Interesting that it works, didnt work on other repos such as knmstate, kubevirt
I think it worth still to consider please to indeed split it to junit-report and artifacts as this is the formal way, and this way in case something changes with ginkgo we will be already adapted,
Is there maybe some trap that copies the file ? if so it can explain it, unless the tests runs from the artifacts directory which can also explain why it works

Reply:
IIRC your way didn't work...

Reply:
well it worked on all other repos, worth another try if you can please

Reply:
I am pretty sure it was not working if you don't specify `--output-dir`, maybe they have fix it at newer version ?

Reply:
donno. didn't find any reference of such a fix..

Reply:
Please see
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_cluster-network-addons-operator/1260/pull-e2e-cluster-network-addons-operator-monitoring-k8s/1511616494230835200
```
+ cp '/tmp/cnao//cluster-network-addons-operator/test_logs/e2e/workflow/*.log' /logs/artifacts
cp: cannot stat '/tmp/cnao//cluster-network-addons-operator/test_logs/e2e/workflow/*.log': No such file or directory
```

Reply:
>using the right args, can remove the need of cp if related

that cp doesn't have to succeed, it has a `|| true`

>P.S panic: test timed out after 10m0s might also be due to this PR ?

think so..

Reply:
The fact that there is `|| true` doesnt surely mean it is allowed to fail, it was there in case there isn't a file imo.
Since a file is not there, it didnt manage to copy it and the junit failed to be displayed.
It smells like it might be because of the output-dir, this is what we saw at nmstate at least

The junit should exists if the test ran, even if they paniced